### ISSUE TITLE
Rerunning an algo

### ISSUE THREAD

**Issue by npezolano** [quantopian/zipline#529]
Is it possible to rerun an algo and continue from a different time period. Take the following simple example, I want to run the algo from t1 to t2, and then run it again from t2 to t3, however the algo's portfolios gets overwritten.

```python
from zipline import TradingAlgorithm
from zipline.utils.factory import load_from_yahoo

from datetime import datetime
import pytz
import matplotlib.pyplot as plt

class BuyApple(TradingAlgorithm):
    def initialize(self):
        self.invested = False
        self.i = 0

    def handle_data(self, data):
        self.order('AAPL',1)
        self.i += 1

# Load data for t1 to t2
start = datetime(2001, 1, 1, 0, 0, 0, 0, pytz.utc)
end = datetime(2002, 1, 1, 0, 0, 0, 0, pytz.utc)
data = load_from_yahoo(stocks=['AAPL'], indexes={}, start=start,
                   end=end, adjusted=False)

# Run algorithm
dma = BuyApple(capital_base=10000000)
perf = dma.run(data)
perf.positions[-1] # 'amount': 247

# Load data for t2 to t3
start = datetime(2002, 1, 2, 0, 0, 0, 0, pytz.utc)
end = datetime(2002, 5, 1, 0, 0, 0, 0, pytz.utc)
data = load_from_yahoo(stocks=['AAPL'], indexes={}, start=start,
                   end=end, adjusted=False)

perf = dma.run(data)
perf.positions[-1] #  'amount': 83
```

**Comment by twiecki** [comment]
That's an interesting use-case. Definitely makes sense to me that you'd want to do that.

One thing to try is to set `overwrite_sim_params=True` in `run()`: https://github.com/quantopian/zipline/blob/master/zipline/algorithm.py#L400

But note that you then would have to specify your own `sim_params` and pass them to `__init__` which set start and end date of the whole length.

**Comment by npezolano** [comment]
Thanks for the response, I tried your approach, the problem with the above approach is that the `PerformanceTracker` gets set to `None` here, https://github.com/quantopian/zipline/blob/master/zipline/algorithm.py#L474

**Comment by twiecki** [comment]
Yes, that's indeed the problem :-/. I don't think we currently allow for this then but there are some plans to rework this logic a bit so we'll keep this in mind to make sure to support it.

**Comment by npezolano** [comment]
This can be closed by #852 .
